File: dev/20190520/k2works/api/src/main/java/k2works/FizzBuzzValueCommand.java
package k2works;

import k2works.domain.type.FizzBuzzType;

public class FizzBuzzValueCommand implements IFizzBuzzCommand {
  private FizzBuzzType _type;

  public FizzBuzzValueCommand(FizzBuzzType type) {
    _type = type;
  }

  @Override
  public IFizzBuzz execute(int number) {
    return _type.generate(number);
  }

}

File: dev/20190822/k2works/src/Application/Command/FizzBuzzValueCommand.ts
import { IFizzBuzzType } from "../../Domain/Type/IFizzBuzzType";
import { IFizzBuzzCommand } from "./IFizzBuzzCommand";

export class FizzBuzzValueCommand implements IFizzBuzzCommand {
  // tslint:disable-next-line: variable-name
  private _type: IFizzBuzzType;

  constructor(type: IFizzBuzzType) {
    this._type = type;
  }
  public execute(
    input: number
  ): import("../../Domain/Model/IFizzBuzz").IFizzBuzz {
    return this._type.generate(input);
  }
}
